### PR TITLE
Batchprocess image

### PR DESCRIPTION
**Pull request by tarujg**

### CODE REVIEW

**Comment by tarujg** [comment]
@ktl014 @igneel73 @zhouyuanyuan1999 
These are the updated scripts for preprocessing.

File: cvtools.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+import os
+from math import pi
+import cv2
+from skimage import morphology, measure, restoration
+from skimage.filters import scharr, gaussian
+import numpy as np
+from scipy import ndimage
+
+
+class configuration:
```
Comment (ktl014) [inline cvtools.py:17]: Import this into config/config.py 
Change all of this to snake case

example of how to use the opt
- say I want to get the ImagesPerDir from the configuration to do my preprocessing:
```
from config import opt
images_per_dir = opt.ImagesPerDir
```

File: pipeline.py [hunk 2]
```diff
@@ -26,10 +31,36 @@ def predict(self):
         #TODO @ZhouyuanYuan
         pass
 
-    def process(self):
-        """Process binary files (not to be confused with model preprocessing"""
-        #TODO @TarujGoyal
-        pass
+    def process(self, date, root="/data6/phytoplankton-db/hab_in_vitro"):
```
Comment (ktl014) [inline pipeline.py:34]: set the default value for `root` by importing a set `data_dir`
Also root_dir should be `/data6/phytoplankton-db/hab_in_vitro/images`
i.e.

```
pipeline.py
# Project level imports
from config.config import Config

# Module level constants
DEFAULT_ENV = CONST.PROD_ENV
opt = Config(DEFAULT_ENV)
...
..
.
def process(self, date, root=opt.data_dir)
```

Comment (ktl014) [reply pipeline.py:34]: also can you set up the main() and set the pipeline.py to accept a date argument and feed that into pipeline.process()

File: spc.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+    image_res = configuration.PixelSize / 1000
+
+    # print("Image resolution is set to: {} mm/pixel.".format(str(image_res)))
+
+    entry = {DBCONST.IMG_FNAME: os.path.join(image_dir, filename),
+             DBCONST.IMG_ID: filename.split('.')[0],
+             DBCONST.IMG_DATE: datetime.date.fromtimestamp(timestamp).strftime('%Y-%m-%d'),
+             DBCONST.IMG_TIME: datetime.datetime.fromtimestamp(timestamp).strftime('%H:%M:%S'),
+             DBCONST.IMG_TSTAMP: timestamp,
+             'file_size': os.path.getsize(image_path) / 1000.0,
```
Comment (ktl014) [inline spc.py:61]: add these 'file_size' to the DBCONST class in constants

File: spc.py [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+             'eccentricity': features['eccentricity'],
+             DBCONST.ORIENT: features['orientation'] * 180 / math.pi,
+             DBCONST.MJR_LEN: features['major_axis_length'] * image_res,
+             DBCONST.MIN_LEN: features['minor_axis_length'] * image_res,
+             DBCONST.HEIGHT: img_c_8bit.shape[0],
+             DBCONST.WIDTH: img_c_8bit.shape[1],
+             'solidity': features['solidity'],
```
Comment (ktl014) [inline spc.py:68]: same for `solidity` and `eccenntricity`

File: spc.py [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+        except:
+            time.sleep(0.05)
+
+    # Terminate the processes in case they are stuck
+    for p in processes:
+        p.terminate()
+
+    print("3. Exporting spreadsheet results...")
+    df = pandas.DataFrame.from_dict(entry_list)
+    df.to_csv(os.path.join(subdir, 'features.tsv'), index=False, sep='\t')
```
Comment (ktl014) [inline spc.py]: can we set it as `.csv` files and use sep=','? this is to be consistent with our current procedurees.

File: spc.py [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+
+    # Insert into database
+    print("4. Inserting into Database")
+    insert_database(df, db_path="test.db", table_name='date_sampled')
```
Comment (ktl014) [inline spc.py:213]: test.db should also be imported from constants.
Lets use the DEV_ENV to be "test.db" then PROD_ENV to be the actual "plankton.DB"

**Review by ktl014 — APPROVED** [review]
Good to go. Just make the changes for the configuration imports, take out hard codings, and set up the main() within pipeline.py.

File: spc.py (hunk 5 above)
Comment (tarujg) [reply spc.py]:
```suggestion
    df.to_csv(os.path.join(subdir, 'features.csv'), index=False, sep=',')
```

**Comment by tarujg** [comment]
These changes have been pushed to the branch directly

**Pull request closed without merging**